dipy/workflows/tests/test_io.py

```diff
@@ -91,3 +92,28 @@ def test_split_flow():
         split_data, split_affine = load_nifti(split_path)
         npt.assert_equal(split_data.shape, volume[..., 0].shape)
         npt.assert_array_almost_equal(split_affine, affine)
+
+
+MockData = MagicMock(name='MockData')
+MockAffine = MagicMock(name='MockAffine')
+MockImage = MagicMock(name='MockImage')
+
+
+def test_convert_sh_flow():
+    with patch(
+        "dipy.workflows.io.convert_sh_descoteaux_tournier"
+    ) as mock_convert, patch(
+        "dipy.workflows.io.load_nifti", return_value=(
+            MockData, MockAffine, MockImage
+        )
+    ), patch(
+        "dipy.workflows.io.save_nifti"
+    ), patch.object(
+        ConvertSHFlow, "get_io_iterator", return_value=iter([(0, 0)])
+    ):
+
+        workflow = ConvertSHFlow()
+        workflow.run("fake_input_path", "fake_output_dir")
+
+        # just test that the convert function gets called
+        mock_convert.assert_called()
```

Comment:
Can you look how we tests our workflows and follow the standard way ? (meaning avoid `mock`)

Thank you

Reply:
The advantage of mocking is that unit tests become more independent. Then when someone breaks something fewer tests fail and the ones that do fail make it more clear what actually broke. 

If I follow the other examples then I have to use some of the example data from `get_fnames` right? I don't think there are there test FOD images saved as nifti, so I am not sure how to follow those examples without at least mocking `load_nifti`. Any advice?

Reply:
> The advantage of mocking is that unit tests become more independent. Then when someone breaks something fewer tests fail and the ones that do fail make it more clear what actually broke.

I understand the advantage. However, to keep consistency and facilitate the maintenance, we try to keep the same standard. When we decided to switch to mock, we will put an effort to change all worflows using mock. Feel free to create an issue to discuss to this point.

> If I follow the other examples then I have to use some of the example data from get_fnames right? I don't think there are there test FOD images saved as nifti, so I am not sure how to follow those examples without at least mocking load_nifti. Any advice?

I will follow your `test_convert_sh_descoteaux_tournier`.   Meaning:
- Do it all in a temporary directory (`with TemporaryDirectory() as tmpdir:`)
- use the same input array (`sh_coeff`), and save it as a nifti, with identity matrix as an affine
- run the workflows
- load the output nifti
- compare the output nifiti with expected_sh_coeff

Does it make sense ?